src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -151,13 +153,32 @@ public Object instanceMainPostParseArgs() {
 
         LoggingUtils.setLoggingLevel(VERBOSITY);  // propagate the VERBOSITY level to logging frameworks
 
-        for (final File f : TMP_DIR) {
+        for (final String s : TMP_DIR) {
+            final Path p = IOUtils.getPath(s);
             // Intentionally not checking the return values, because it may be that the program does not
             // need a tmp_dir. If this fails, the problem will be discovered downstream.
-            if (!f.exists()) f.mkdirs();
-            f.setReadable(true, false);
-            f.setWritable(true, false);
-            System.setProperty("java.io.tmpdir", f.getAbsolutePath()); // in loop so that last one takes effect
+            if (!Files.exists(p)) {
+                try {
+                    Files.createDirectories(p);
+                } catch (IOException e) {
+                    // intentionally ignoring
```

Comment:
Don't create the temp dir if it doesn't exist -- just throw a `UserException` in this case

Reply:
I personally like specifying a non-existent directory for temporary files that is created by the program instead of explicitly using `mkdirs` - anyway, I can change in the next commit.

Reply:
Removed in the next commit.